packages/api/src/decorator/private/useActivityDecoratorRequest.ts

```diff
@@ -0,0 +1,24 @@
+import { WebChatActivity, getActivityLivestreamingMetadata } from 'botframework-webchat-core';
+import { useMemo } from 'react';
+import { ActivityDecoratorRequest } from '..';
+
+const supportedActivityRoles: ActivityDecoratorRequest['from'][] = ['bot', 'channel', 'user', undefined];
+
+export default function useActivityDecoratorRequest(activity: WebChatActivity) {
+  return useMemo<ActivityDecoratorRequest>(() => {
+    const { type } = getActivityLivestreamingMetadata(activity) || {};
+
+    return {
+      activity,
+      livestreamingState:
```

Comment:
I think we should let web devs to compute this value from `activity` themselves.